x11-libs/xcb-imdkit/xcb-imdkit-1.0.5.ebuild

```diff
@@ -0,0 +1,36 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="Input method development support for xcb"
+HOMEPAGE="https://github.com/fcitx/xcb-imdkit"
+SRC_URI="https://download.fcitx-im.org/fcitx5/${PN}/${P}.tar.xz"
+
+LICENSE="LGPL-2.1 !system-uthash? ( BSD-1 )"
+SLOT="5"
+KEYWORDS="~amd64 ~arm64 ~loong ~riscv ~x86"
+IUSE="+system-uthash"
+
+RDEPEND="
+	x11-libs/libxcb
+	x11-libs/xcb-util
+	x11-libs/xcb-util-keysyms
+"
+DEPEND="
+	${RDEPEND}
+	system-uthash? ( dev-libs/uthash )
+"
+BDEPEND="
+	kde-frameworks/extra-cmake-modules:0
+	virtual/pkgconfig
+"
+
+src_configure() {
+	local mycmakeargs=(
+		-DUSE_SYSTEM_UTHASH=$(usex system-uthash ON OFF)
```

Comment:
I think we should always use the system library instead of relying on the bundled one.

Reply:
I have removed system-uthash use, and moved dev-libs/uthash to BDEPEND.

Reply:
> DEPEND not BDEPEND

uthash only contains header files, it should only be needed at compile time.
So I think it should belong to BDEPEND. did I misunderstand somewhere?

Reply:
DEPEND and BDEPEND are both for compile only dependencies.

See CHOST v CBUILD and the explanation for the BDEPEND/DEPEND split.

https://devmanual.gentoo.org/general-concepts/dependencies/index.html#chost-vs-cbuild
https://devmanual.gentoo.org/general-concepts/dependencies/index.html#build-dependencies